Update/information box typography

enejb opened the pull request:
Fixes #953

After:
Placeholder 
![screen shot 2015-12-15 at 15 30 00](https://cloud.githubusercontent.com/assets/115071/11827402/e1f94a8c-a340-11e5-91ae-9345ace295e9.png)

Multi site view.
![screen shot 2015-12-15 at 16 12 27](https://cloud.githubusercontent.com/assets/115071/11828153/c126f3da-a346-11e5-95a1-60812683554e.png)

Single site view.
![screen shot 2015-12-15 at 15 30 16](https://cloud.githubusercontent.com/assets/115071/11827404/e4280ece-a340-11e5-95af-93f851421124.png)

Mobile view
![screen shot 2015-12-15 at 16 25 03](https://cloud.githubusercontent.com/assets/115071/11828338/a513f83a-a348-11e5-8bb3-823050c801b4.png)

Mobile view:
![screen shot 2015-12-15 at 16 24 47](https://cloud.githubusercontent.com/assets/115071/11828343/a8940f7c-a348-11e5-99bb-d27fa05ac650.png)

Single Site view:
A plugin that doesn't have much info.
![screen shot 2015-12-15 at 16 13 01](https://cloud.githubusercontent.com/assets/115071/11828156/c457dfd8-a346-11e5-865f-8756aa4404b8.png)

Business Plugin:
![screen shot 2015-12-15 at 16 31 25](https://cloud.githubusercontent.com/assets/115071/11828425/7f678254-a349-11e5-8d3e-39082b33e443.png)


![screen shot 2015-12-15 at 16 08 51](https://cloud.githubusercontent.com/assets/115071/11828108/5de2ad64-a346-11e5-9242-c323f6d1f34b.png)




**To test**
Visit a single plugin page. For example 
http://calypso.localhost:3000/plugins/akismet and also http://calypso.localhost:3000/plugins/akismet/example.com

Browse plugins single plugin pages that have comptible versions with wp. 
Haven't been updated in a long time. 
Need an update.
Etc. 
Does it look and function as expected? 

cc: @rickybanister, @johnHackworth  and @lezama

johnHackworth commented on client/my-sites/plugins/plugin-information/index.jsx:
```diff
@@ -81,52 +78,68 @@ export default React.createClass( {
+			const dateFromNow = i18n.moment.utc( this.props.plugin.last_updated, 'YYYY-MM-DD hh:mma' ).fromNow();
+			const syncIcon = this.props.hasUpdate ? <Gridicon icon="sync" size={ 18 } /> : null;
+
+			return <div className="plugin-information__last-updated">
```
maybe 
```
return (
    <div> 
        ...
    </div>
);
```

johnHackworth commented on client/my-sites/plugins/plugin-information/index.jsx:
```diff
@@ -153,24 +166,22 @@ export default React.createClass( {
-							{ this.renderStarRating() }
-							{ this.renderRatingsDetail() }
+							<PluginRatings
+								rating={ this.props.plugin.rating }
```
being a placeholder, wouldn't it make more sense to pass a hardcorded value (any) in all this properties? Since they are undefined it looks like this part is not being rendered at all

enejb replied:
Yeah I still have to give the placeholders a lot more love.
I just wanted to get the review process started. :)

johnHackworth commented:
Looks superb! And codewise is mostly good... I think one placeholder is not working as it should (I may be wrong, though), but it's almost ready to go

rickybanister commented:
@enejb I noticed a few small issues with the layout/type:

<img width="368" alt="screen shot 2015-12-11 at 10 17 34 am" src="https://cloud.githubusercontent.com/assets/437258/11747303/747f6f5c-9ff0-11e5-9b5c-c72f725c740a.png">

The version info shouldn't have a min-width on it—that won't be sustainable. Can we do a flex or a table layout for this area? We just want the compatibility info to be left-aligned and padded off the version info, not with a min-width. Also the line-height is different so the text doesn't land on a baseline grid.

rickybanister commented:
Also can we make the trashcan center on the toggle the same way I updated the (i) info icon?

The label should right-align with the other labels and the can should be centered within the 24px width of the toggle. It may need a `margin-top: -1px` since it's 18px tall and the toggles are 16px—not sure.

<img width="128" alt="screen shot 2015-12-11 at 10 20 22 am" src="https://cloud.githubusercontent.com/assets/437258/11747416/2028272c-9ff1-11e5-966c-b38bc1e441ba.png">

enejb commented:
@rickybanister Can you take a look please again. 
- I have implemented a table type layout for a single site view and a non table layout for the multi site view. 
- Improved line-height.
- I updated the trashcan so it better aligned. 
- Improved the placeholder layout. 

cc @johnHackworth for a code review.

rickybanister commented:
I'm going to check this out and take a closer look, but I notice that the `setup` external icon has the same issue as the trashcan had—all of those icons should be centered in the 24px width of the toggle and the labels should all align to the right.

johnHackworth commented on client/my-sites/plugins/plugin-information/index.jsx:
```diff
@@ -62,16 +61,14 @@ export default React.createClass( {
-					<Gridicon size={ 12 } icon="external" />{ this.translate( 'WordPress.org Plugin page' ) }
-				</a>
-			</div>
+			<ExternalLink
```
it's so good to see all that html structures get replaced by shared components... :)

johnHackworth commented:
code-wise this looks pretty good to me :+1:

rickybanister commented:
@enejb I pushed a pretty big refactor to the layout of the plugins-information and the styles. There is still probably a little bit to clean up, but I think I achieved a simpler layout via using flex.

I got tripped up because the placeholder is separate from the actual render and was editing the wrong one. Would you mind taking a look at the placeholder and making sure the layout reflects the new layout of the actual render? I can then adjust the styles again if needed. 

Sorry for the big change, but I think it'll help us in the long-run to use flex more consistently.

enejb commented:
The Placeholders currently look like this

Desktop:
![screen shot 2015-12-16 at 16 21 41](https://cloud.githubusercontent.com/assets/115071/11857903/454b5744-a411-11e5-981b-e1cba328f2e3.png)

and Mobile:
![screen shot 2015-12-16 at 16 22 25](https://cloud.githubusercontent.com/assets/115071/11857904/4a9e4eae-a411-11e5-8208-4cb30ecbe37d.png)

enejb commented:
Just updated them to look like this:
Mobile:
![screen shot 2015-12-16 at 16 40 33](https://cloud.githubusercontent.com/assets/115071/11858166/ed1e4574-a413-11e5-8222-9c50e5ff7bec.png)

Desktop:
![screen shot 2015-12-16 at 16 40 21](https://cloud.githubusercontent.com/assets/115071/11858168/efaea0a4-a413-11e5-9e3a-9ceac3a3a299.png)

rickybanister commented:
My placeholder doesn't look like that @enejb, did you push everything?

<img width="894" alt="screen shot 2015-12-17 at 11 01 19 am" src="https://cloud.githubusercontent.com/assets/437258/11874417/89da7e20-a4ad-11e5-9763-cf76440c2658.png">

I might have something strange going on, but if we can make it look like your screenshot this is ready to merge.

enejb commented:
@rickybanister I updated the placeholders to be more like the previous version Which I think look better.
![screen shot 2015-12-17 at 13 45 37](https://cloud.githubusercontent.com/assets/115071/11883553/a332af94-a4c8-11e5-980a-feaca4d16118.png)

rickybanister commented:
:ship:

johnHackworth commented:
I've checked this both code wise and functional-wise and I think it's ready to go
